homeassistant/components/stream/__init__.py

```diff
@@ -92,98 +91,15 @@
     "RTSP_TRANSPORTS",
     "SOURCE_TIMEOUT",
     "Stream",
+    "StreamClientError",
+    "StreamOpenClientError",
     "create_stream",
     "Orientation",
 ]
 
 _LOGGER = logging.getLogger(__name__)
 
 
-class StreamClientError(IntEnum):
-    """Enum for stream client errors."""
-
-    BadRequest = 400
-    Unauthorized = 401
-    Forbidden = 403
-    NotFound = 404
-    Other = 4
-
-
-class StreamOpenClientError(HomeAssistantError):
-    """Raised when client error received when trying to open a stream.
-
-    :param stream_client_error: The type of client error
-    """
-
-    def __init__(
-        self, *args: Any, stream_client_error: StreamClientError, **kwargs: Any
-    ) -> None:
-        self.stream_client_error = stream_client_error
-        super().__init__(*args, **kwargs)
-
-
-async def _async_try_open_stream(
-    hass: HomeAssistant, source: str, pyav_options: dict[str, str] | None = None
-) -> InputContainer | OutputContainer:
-    """Try to open a stream.
-
-    Will raise StreamOpenClientError if an http client error is encountered.
-    """
-    return await hass.loop.run_in_executor(None, _try_open_stream, source, pyav_options)
-
-
-def _try_open_stream(
-    source: str, pyav_options: dict[str, str] | None = None
-) -> InputContainer | OutputContainer:
```

Comment:
This is changed to only be `InputContainer` when moving since we're only using one of the open constructors for input.